test/jdk/java/net/httpclient/HttpRequestBuilderTest.java

```diff
@@ -378,6 +384,38 @@ public static <R,P1, P2> R test2(String name, R receiver, BiFunction<P1, P2, R>
         }
     }
 
+    private static void verifyCopy() {
+
+        // Create the request builder
+        HttpRequest.Builder requestBuilder = HttpRequest
+                .newBuilder(TEST_URI)
+                .header("X-Foo", "1")
+                .method("GET", noBody())
+                .expectContinue(true)
+                .timeout(Duration.ofSeconds(0xBEEF))
+                .version(HttpClient.Version.HTTP_2);
+
+        // Create the original and the _copy_ requests
+        HttpRequest request = requestBuilder.build();
+        HttpRequest copiedRequest = requestBuilder
+                .copy()
+                .header("X-Foo", "2")
+                .header("X-Bar", "3")
+                .build();
+
+        // Verify copied _references_
+        assert request.uri().equals(copiedRequest.uri());
```

Comment:
Hello Volkan, within the JDK build, we launch these jtreg tests with assertions enabled by default. However, from what I see in the build files, that's configurable. So it may not always be the case that assertions are enabled for these tests.

Furthermore, I don't remember seeing `assert` being used within the test code to verify the test expectations. So I think it would be better to do a conditional check here and explicitly throw a `AssertionError` if that check fails, like we do in some other places of this test.

Reply:
Fixed in 5eddbf7d4477cf997111b4dae8b3cb112216fadc.